app/lib/dns.server.ts

```diff
@@ -10,14 +11,28 @@ import type {
   GetChangeResponse,
 } from '@aws-sdk/client-route-53';
 import { isIPv4, isIPv6 } from 'is-ip';
+import { secrets } from 'docker-secret';
+
+declare global {
```

Comment:
I don't think we need to do this, using `process.env` is fine for this.

Reply:
What you could do is have this module throw if the hosted zone id isn't configured in the env, and then assume it exists for the rest of the methods below.

Reply:
Should I remove `global var`? I still have to create a hosted zone dynamically and supply `hostedZoneId` to the module for the test. I haven't found a way to feed `hostedZoneId` other than using `global var`. Would you advise how I achieve it?

Reply:
Yeah, I'd remove it.  Probably you're right, we should always pass it in as an arg (first arg) to all these functions at this level, and at higher levels of the API, use the value on the environment.

Reply:
I've been struggling with this concept. I will push my update soon. I hope I understand the concept here.

Reply:
I've been trying not to use `global var` while removing unnecessary parameter, but it seems not doable for me now. Here's my thoughts. 

`createUserDomain()` in `domains.server` calls `createRecord()`. So both `domains`, `dns` level functions need `hostedZoneId` as it's generated dynamically in testing. 

As per knowledge now, I can use `global var` and remove `hostedZoneId` parameter in all functions or add the parameter to both `dns` and `domains` level. Please let me know if I'm missing something here.

Reply:
oh, I thought `process.env` is constant. So much learning from real world problem solving. Thank you!